File: D01.jl
using DelimitedFiles, Combinatorics
input = readdlm("D01.txt", Int)

D01(input, n) =
    reduce(*, filter(x -> sum(x) == 2020, collect(combinations(input, n)))[1])

println(D01(input, 2))
println(D01(input, 3))

File: D01.rb
input = File.read("D01.txt").split().map(&:to_i)

def D01(input, n)
    input.combination(n).select { |x| x.sum == 2020}[0].reduce(:*)
end

puts D01(input, 2)
puts D01(input, 3)
